src/ucs/datastruct/ptr_array.c

```diff
@@ -62,8 +81,9 @@ static void UCS_F_MAYBE_UNUSED ucs_ptr_array_dump(ucs_ptr_array_t *ptr_array)
     ucs_trace_data("ptr_array start %p size %u", ptr_array->start, ptr_array->size);
     for (i = 0; i < ptr_array->size; ++i) {
         if (ucs_ptr_array_is_free(ptr_array, i)) {
-            ucs_trace_data("[%u]=<free> (%u)", i,
-                           ucs_ptr_array_placeholder_get(ptr_array->start[i]));
+            ucs_trace_data("[%u]=<free> [%u]=<next> (%u)", i,
```

Comment:
were you going to print num of free elements for every index, like below?
```
ucs_trace_data("[%u]=<free>  (%u)  [%u]=<next>", i,
                          ucs_ptr_array_size_free_get(ptr_array->start[i]),
                          ucs_ptr_array_freelist_get_next(ptr_array->start[i]));
```

Reply:
Yes, well, the number of free elements is actually the number of free elements ahead of the current element - So yes we are printing it in the dump function... 
This number is maintained per element in the free list and does the trick of fast skipping of free elements during the foreach operation.

Reply:
yes, I mean that you probably want this format
`[%u]=<free>  (%u)  [%u]=<next>",`   rather than this
`"[%u]=<free> [%u]=<next> (%u)"`